lib/ui/text/paragraph_builder.cc

```diff
@@ -526,6 +526,7 @@ Dart_Handle ParagraphBuilder::addPlaceholder(double width,
 
 void ParagraphBuilder::build(Dart_Handle paragraph_handle) {
   Paragraph::Create(paragraph_handle, m_paragraphBuilder->Build());
+  m_paragraphBuilder->Reset();
```

Comment:
Is there potential performance downside to always calling `Reset` after building a paragraph even if the `ParagraphBuilder` will not be reused?

In addition to clearing or swapping various structures, the `Reset` may do some slightly more expensive operations such as SkParagraph `ParagraphBuilderImpl::setParagraphStyle` copying various text styles.

If the framework knows when it will be reusing a `ParagraphBuilder`, then I'd prefer to have the framework call `Reset` only when needed.

Reply:
I have to keep measuring this. My initial measurements looked promising but now I'm having trouble showing that it's not all due to noise. You might be right about this - I'm going to add some more tracing locally to see if I can figure out what's going on here.

Reply:
I have a patch for Skia that would make the reset call cheaper - would you still be concerned in that case?